packages/components/src/highlight-cards/weekly-highlight-cards.tsx

```diff
@@ -40,11 +40,100 @@ type WeeklyHighlightCardsProps = {
 	currentPeriod: string;
 };
 
+type HighlightCardsSettingsProps = {
+	onTogglePeriod: ( period: string ) => void;
+	currentPeriod: string;
+};
+
 export const PAST_SEVEN_DAYS = 'past_seven_days';
 export const PAST_THIRTY_DAYS = 'past_thirty_days';
 export const BETWEEN_PAST_EIGHT_AND_FIFTEEN_DAYS = 'between_past_eight_and_fifteen_days';
 export const BETWEEN_PAST_THIRTY_ONE_AND_SIXTY_DAYS = 'between_past_thirty_one_and_sixty_days';
 
+const HighlightCardsSettings = function ( {
+	currentPeriod,
+	onTogglePeriod,
+}: HighlightCardsSettingsProps ) {
+	const translate = useTranslate();
+
+	// @TODO: Fetch the state from API to determine whether showing the settings tooltip.
+	const showSettingsTooltip =
+		sessionStorage.getItem( 'jp-stats-settings-tooltip' ) === '1' ? false : true;
+
+	const settingsActionRef = useRef( null );
+	const [ isSettingsTooltipVisible, setSettingsTooltipVisible ] = useState( showSettingsTooltip );
+	const [ isPopoverVisible, setPopoverVisible ] = useState( false );
+
+	// @TODO: Update the state when users dismiss the settings tooltip.
+	const dismissSettingsTooltip = useCallback( () => {
+		sessionStorage.setItem( 'jp-stats-settings-tooltip', '1' );
+		setSettingsTooltipVisible( false );
+	}, [] );
+
+	// @TODO: Set the popover to disappear when the users click outside of the popover.
```

Comment:
Given the large size of the Jetpack Stats page, I'm unsure if we want to do this.

Reply:
Could you provide more information about the issue between listening to the mouse event and the page size?

Reply:
The main thing I'd like for us to avoid is letting the tooltip be dismissed by accident or without the user being aware that they've done so (e.g., by clicking on other interactive parts of the page or clicking on some whitespace on the page when the tooltip isn't visible).

It's currently a hypothetical concern, so I'll mark this resolved 🙂

Reply:
> letting the tooltip be dismissed by accident or without the user being aware that they've done so

I understand. The original intention is to mitigate users' UI operation to dismiss the settings Popover on which we have a different level of significance definition. Let's talk about it as an improvement later. 🙂

BTW, I found the [built-in functionality](https://github.com/Automattic/wp-calypso/blob/trunk/packages/components/src/popover/index.jsx#L163C1-L180) of the Popover component to dismiss when clicking outside.